crates/starknet/src/starknet/dump.rs

```diff
@@ -0,0 +1,178 @@
+                    let mut file = OpenOptions::new()
+                        .append(true)
+                        .write(true)
+                        .open(file_path)
+                        .map_err(Error::IoError)?;
+                    let lenght = file.seek(SeekFrom::End(0)).map_err(Error::IoError)?;
+                    file.set_len(lenght - 1).map_err(Error::IoError)?; // remove last "]" with set_len
+                    file.write_all(format!(", {transaction_dump}]").as_bytes())
+                        .map_err(Error::IoError)?;
+                } else {
+                    // create file
+                    let transactions = vec![transaction];
+                    let transactions_dump = serde_json::to_string(&transactions).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    fs::write(Path::new(&path), transactions_dump)?;
+                }
+
+                Ok(())
+            }
+            None => Err(Error::FormatError),
+        }
+    }
+
+    /// save starknet transactions to file
+    pub fn dump_transactions(&self) -> DevnetResult<()> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let transactions = &self
+                    .transactions
+                    .iter()
+                    .map(|x| x.1.inner.clone())
+                    .collect::<Vec<Transaction>>();
+
+                // dump only if there are transactions to dump
+                if !transactions.is_empty() {
+                    let transactions_dump = serde_json::to_string(transactions).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    fs::write(Path::new(&path), transactions_dump)?;
+                }
+
+                Ok(())
+            }
+            None => Err(Error::FormatError),
+        }
+    }
+
+    // load starknet transactions from file
+    pub fn load_transactions(&self) -> DevnetResult<Vec<Transaction>> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let file_path = Path::new(path);
+
+                // load only if the file exists, if dump_path is set but the file doesn't exist it
+                // can mean that it's first run with dump_path parameter set to dump, in that case
+                // return empty vector
+                if file_path.exists() {
+                    let mut file = File::open(file_path).map_err(Error::IoError)?;
+                    let mut data = String::new();
+                    file.read_to_string(&mut data).map_err(Error::IoError)?;
+                    let transactions: Vec<Transaction> =
+                        serde_json::from_str(&data).map_err(|_| Error::DeserializationError {
+                            obj_name: "Vec<Transaction>".to_string(),
+                        })?;
```

Comment:
Should rely on `serde_json::from_reader(file)`

Reply:
fixed 71e7e3d3d62d9c041f84c577a9d748c15f66a79a